src/AutoRest.CSharp/LowLevel/Output/LowLevelClient.cs

```diff
@@ -121,33 +125,97 @@ public static IEnumerable<LowLevelClientMethod> BuildMethods(RestClientBuilder b
+                .ToArray();
 
             if (Fields.CredentialFields.Count == 0)
             {
-                yield return BuildPublicConstructor(null, clientOptionsParameter);
+                yield return CreateSecondaryConstructor(requiredParameters, optionalParametersArguments);
             }
             else
             {
                 foreach (var credentialField in Fields.CredentialFields)
                 {
-                    yield return BuildPublicConstructor(credentialField, clientOptionsParameter);
+                    yield return CreateSecondaryConstructor(requiredParameters.Append(CreateCredentialParameter(credentialField!.Type)).ToArray(), optionalParametersArguments);
                 }
             }
         }
 
-        private ConstructorSignature BuildPublicConstructor(FieldDeclaration? credentialField, Parameter clientOptionsParameter)
+        private ConstructorSignature CreatePrimaryConstructor(IReadOnlyList<Parameter> parameters)
+            => new(Declaration.Name, $"Initializes a new instance of {Declaration.Name}", Public, parameters);
+
+        private ConstructorSignature CreateSecondaryConstructor(IReadOnlyList<Parameter> parameters, FormattableString[] optionalParametersArguments)
         {
-            var constructorParameters = RestClientBuilder.GetConstructorParameters(Parameters, credentialField?.Type).Append(clientOptionsParameter).ToArray();
-            return new ConstructorSignature(Declaration.Name, $"Initializes a new instance of {Declaration.Name}", Public, constructorParameters);
+            var arguments = parameters
+                .Select<Parameter, FormattableString>(p => $"{p.Name}")
+                .Concat(optionalParametersArguments)
+                .ToArray();
+            return new(Declaration.Name, $"Initializes a new instance of {Declaration.Name}", Public, parameters, new ConstructorInitializer(false, arguments));
+        }
+
+        private Parameter CreateCredentialParameter(CSharpType type)
+        {
+            return new Parameter(
+                "credential",
+                "A credential used to authenticate to an Azure Service.",
+                type,
+                null,
+                true);
+        }
+
+        private ConstructorSignature CreateMockingConstructor()
+            => new(Declaration.Name, $"Initializes a new instance of {Declaration.Name} for mocking.", Protected, Array.Empty<Parameter>());
+
+        private Parameter CreateOptionsParameter()
+        {
+            var clientOptionsType = ClientOptions.Type.WithNullable(true);
+            return new Parameter("options", "The options for configuring the client.", clientOptionsType, Constant.NewInstanceOf(clientOptionsType), false)
+            {
+                forceInitializeValue = Constant.NewInstanceOf(clientOptionsType)
```

Comment:
Initializer for the parameter should've been written by this line: https://github.com/Azure/autorest.csharp/blob/feature/v3/src/AutoRest.CSharp/Common/Generation/Writers/CodeWriterExtensions.cs#L253

Why `parameter.DefaultValue != null && !TypeFactory.CanBeInitializedInline(parameter.Type, parameter.DefaultValue)` returns `false` for this parameter?

Reply:
I added this because options is optional because of this: https://github.com/Azure/autorest.csharp/pull/2128#discussion_r840860585

`options` is required so the condition returns `false`.

Reply:
`options` isn't required - you set `DefaultValue` to `Constant.NewInstanceOf(clientOptionsType)`. Here is the test that passes on my machine:
```cs
[Test]
public void ValidateOptions()
{
    var clientOptionsType = new CSharpType(typeof(AccessibilityClientOptions)).WithNullable(true);
    var parameter = new Parameter("options", "The options for configuring the client.", clientOptionsType, Constant.NewInstanceOf(clientOptionsType), false);
    Assert.IsTrue(parameter.DefaultValue != null && !TypeFactory.CanBeInitializedInline(parameter.Type, parameter.DefaultValue));
}
```

If you need to avoid setting it to null in constructor declaration, you can set `UseDefaultValueInCtorParam: false` (name is misleading though, I need to update it).

Reply:
This is not going to work. Here "`options` is required" I mean we force it as required by 
```C#
var optionalToRequired = optionalParameters.Select(parameter => parameter with { DefaultValue = null }).ToArray();
```
when we `BuildPrimaryConstructors`. We lost default value information here. So we need any other places to store this information, otherwise we don't know what type we should use to initialize the `options`

Reply:
You can simply skip it:
```cs
optionalParameters.Select(p => ClientOptions.Type.EqualsIgnoreNullable(p.Type) ? p : p with { DefaultValue = null }).ToArray()
```

Or you can even pass `ClientOptions` parameter into the method. Something like:
```cs
var requiredParameters = RestClientBuilder.GetRequiredParameters(orderedParameters).ToArray();
var optionalParameters = RestClientBuilder.GetOptionalParameters(orderedParameters).ToArray();
var clientOptionsParameter = CreateOptionsParameter();
return (
    BuildPrimaryConstructors(requiredParameters, optionalParameters, clientOptionsParameter).ToArray(),
    BuildSecondaryConstructors(requiredParameters, optionalParameters, clientOptionsParameter).ToArray()
);
```
and
```cs
private IEnumerable<ConstructorSignature> BuildPrimaryConstructors(IReadOnlyList<Parameter> requiredParameters, IReadOnlyList<Parameter> optionalParameters, Parameter clientOptionsParameter)
{
    var optionalToRequired = optionalParameters
        .Select(parameter => parameter with { DefaultValue = null })
        .Append(clientOptionsParameter)
        .ToArray();
```

```cs
private IEnumerable<ConstructorSignature> BuildSecondaryConstructors(IReadOnlyList<Parameter> requiredParameters, IReadOnlyList<Parameter> optionalParameters, Parameter clientOptionsParameter)
{
    if (requiredParameters.Any() || Fields.CredentialFields.Any())
    {
        yield return CreateMockingConstructor();
    }

    var optionalParametersArguments = optionalParameters
        .Append(clientOptionsParameter)
        .Select(p => p.Type.EqualsIgnoreNullable(typeof(Uri)) ? $"new {typeof(Uri):I}({p.DefaultValue!.Value.GetConstantFormattable()})" : p.DefaultValue!.Value.GetConstantFormattable())
        .ToArray();
```

Reply:
This cannot work either. In this case `options` will be optional.
Here is the problem:
1. We want `options` to be required, therefore the `DefaultValue` should be `null`.
2. We want `options` can be initialized if user passes in `null` as `options ??= new ClientOptions();`. But we don't know what type should be intialized bacause we get this type from `DefaultValue` which is `null` now.

Reply:
It will work with `options` being optional if you set `UseDefaultValueInCtorParam` to false:
https://github.com/Azure/autorest.csharp/blob/feature/v3/src/AutoRest.CSharp/Common/Generation/Writers/CodeWriterExtensions.cs#L203

This way there will be no `= default` generated in the signature.

Reply:
If you want `DefaultValue` not to be `null`, then it will goes to https://github.com/Azure/autorest.csharp/blob/d51e8d41dffbcb1071dfec5b65b24f99194cc464/src/AutoRest.CSharp/Common/Generation/Writers/CodeWriterExtensions.cs#L224, we still have the `= default` in signature.
If you don't want `DefaultValue` not to be `null`, then we need some place to store the initialization type.

Reply:
Ok, so I think it is time to separate `DefaultValue` constant and `Initializer` expression, make `CodeWriterExtensions` just write `DefaultValue` in the signature and `Initializer` in method body without any heuristics. Code that instantiates `Parameter` should fully responsible for what is needed, so there is no need to "adapt" to `CodeWriterExtensions` internal logic.
Here is the [issue](https://github.com/Azure/autorest.csharp/issues/2208) and the [PR](https://github.com/Azure/autorest.csharp/pull/2200).